src/language/mixins.md

```diff
@@ -26,10 +26,17 @@ class Maestro extends Person [!with Musical, Aggressive, Demented!] {
 }
 {% endprettify %}
 
-To _implement_ a mixin, create a class that extends Object and
-declares no constructors.
-Unless you want your mixin to be usable as a regular class,
-use the `mixin` keyword instead of `class`.
+To _implement_ a mixin, use the `mixin` declaration. Or, if you want your mixin
+to be usable as both a regular class and a mixin, use the `mixin class` declaration.
+
+{{site.alert.version-note}}
+  The `mixin class` declaration requires a language version of at least 3.0.
+  See [`class`, `mixin`, or `mixin class`?][] for more information.
+{{site.alert.end}}
+
+Mixins and mixin classes must extend `Object` 
```

Comment:
It's actually a compile-time error for a mixin class declaration to have an extends clause: https://github.com/dart-lang/language/blob/master/accepted/future-releases/class-modifiers/feature-specification.md#:~:text=has%20an%20extends%20clause%2C.

Perhaps:
```suggestion
Mixins and mixin classes cannot have an extends clause. It does not make
sense for a mixin, and for the mixin class it means that its superclass is `Object`.
```

Reply:
That's kind of what I meant, as in extending Object is what classes do implicitly so having an extends clause would mean it's not extending Object. But yes, this is much, much more clear 😅